engine/src/graphics/device.cpp

```diff
@@ -180,4 +234,37 @@ namespace gen
 		return indices;
 	}
 
+	SwapChainSupportDetails GraphicsDevice::querySwapChainSupport(vk::PhysicalDevice device, vk::SurfaceKHR surface)
+	{
+		SwapChainSupportDetails details;
+		details.capabilities		  = device.getSurfaceCapabilitiesKHR(surface);
+		details.formats				  = device.getSurfaceFormatsKHR(surface);
+		details.availablePresentModes = device.getSurfacePresentModesKHR(surface);
+		details.selectedPresentMode	  = chooseSwapPresentMode(details.availablePresentModes, desiredPresentMode_v);
+		return details;
+	}
+
+	vk::SurfaceFormatKHR GraphicsDevice::chooseSwapSurfaceFormat(const std::vector<vk::SurfaceFormatKHR> & availableFormats)
+	{
+		for (const auto & availableFormat : availableFormats)
+		{
+			if (availableFormat.colorSpace != vk::ColorSpaceKHR::eSrgbNonlinear) { continue; }
+
+			if (std::ranges::find(desiredSrgbFormats_v, availableFormat.format) != desiredSrgbFormats_v.end()) { return availableFormat; }
+		}
+
+		// Currently we only support sRGB formats, so if we can't find one, we'll just throw an error.
+		throw gen::vulkan_error("Failed to find suitable surface format that supports SRGB!");
+	}
+
+	vk::PresentModeKHR GraphicsDevice::chooseSwapPresentMode(const std::vector<vk::PresentModeKHR> & availablePresentModes, vk::PresentModeKHR preferredMode)
```

Comment:
Nit: we should ideally support a "priority list"; at least check for `eFifoRelaxed` before defaulting to `eFifo`. `eMailbox` is generally not available on Linux with Nvidia cards (🖕🏽 ).